src/Action/Task/FailureReasons.php

```diff
@@ -18,36 +21,89 @@ public function __construct(
     )
     { }
 
-    private function getDefaultReasons(): array
+    private function loadFailureReasonsConfig(): array
     {
         $path = realpath(__DIR__ . '/../../Resources/config/failure_reasons.yml');
         $parser = new YamlParser();
-        $config = $parser->parseFile($path, Yaml::PARSE_CONSTANT);
+        return $parser->parseFile($path, Yaml::PARSE_CONSTANT);
+    }
+
+    private function getDefaultReasons(): array
+    {
+        $config = $this->loadFailureReasonsConfig();
         return array_map(function($failure_reason) {
             return [
                 'code' => $failure_reason['code'],
                 'description' => $this->translator->trans($failure_reason['description'])
             ];
-        }, $config['failure_reasons']);
+        }, $config['failure_reasons']['default']);
+    }
+
+    private function getTransporterReasons(string $transporter): array
+    {
+        $config = $this->loadFailureReasonsConfig();
+        return array_map(function($failure_reason) {
+            return [
+                'code' => $failure_reason['code'],
+                'description' => $this->translator->trans($failure_reason['description']),
+                'option' => $failure_reason['option'] ?? null,
+                'only' => $failure_reason['only'] ?? null
+            ];
+        }, $config['failure_reasons'][$transporter]);
+    }
+
+    private function getFailureReasons(
+        CustomFailureReasonInterface $entity,
+        bool $transporter
+    ): array
+    {
+        if (
+            $transporter &&
+            $entity instanceof Store &&
+            $entity->isTransporterEnabled()
+        ) {
+            //TODO: Support multi transporter
+            return $this->getTransporterReasons('dbschenker');
+        }
+        $set = $entity->getFailureReasonSet();
+        if (is_null($set)) {
+            return $this->getDefaultReasons();
+        }
+        return $set->getReasons();
     }
 
     public function __invoke($data, Request $request)
     {
 
         $org = $data->getOrganization();
 
+        $transporter = boolval($request->get('transporter', false));
+
         if (is_null($org)) {
             return $this->getDefaultReasons();
         }
 
-        $store = $this->em->getRepository(Store::class)->findOneBy([
-            'organization' => $org
-        ]);
+        $reverse = $this->em->getRepository(Organization::class)
+            ->reverseFindByOrganizarionID($org);
 
-        if (is_null($store->getFailureReasonSet())) {
+        if (empty($reverse)) {
             return $this->getDefaultReasons();
         }
+        $reverse = $reverse[0];
```

Comment:
Using `getOneOrNullResult` in the implementation of `reverseFindByOrganizarionID` would avoid accessing the first element of the array